framework/include/GameTypes.h

```diff
@@ -0,0 +1,60 @@
+#ifndef GAMETYPES_H
+#define GAMETYPES_H
+
+#include <string>
+#include <map>
+
+namespace Game {
+
+// Definition for a move result (success/failure)
+typedef uint8_t moveresult_t;
+enum MoveResult : moveresult_t
+{
+    MOVE_INVALID = 0,
+    MOVE_SUCCESS = 1,
+    MOVE_SUCCESS_GO_AGAIN = 2
+};
+
+// Definition of board result
+typedef int8_t boardresult_t;
+enum BoardResult : boardresult_t
+{
+    GAME_ACTIVE = 0,
+    GAME_OVER_PLAYER1_WIN = 1,
+    GAME_OVER_PLAYER2_WIN = 2,
+    GAME_OVER_TIE = 3
+};
+
+}
+
+namespace Player {
+
+typedef uint8_t playernum_t;
+enum PlayerNumber : playernum_t
```

Comment:
Since this is moved here, the readme may need to be updated to say that the framework only works with two player games.

Reply:
agree. updated the README in the latest commit.